index.js

```diff
@@ -3073,38 +3087,78 @@
   //.
   //. > S.last ([])
   //. Nothing
+  //.
+  //. > S.last (Cons ('baz') (Cons ('foo') (Nil)))
+  //. S.Just ('foo')
+  //.
+  //. > S.last (Nil)
+  //. S.Nothing
   //. ```
-  function last(xs) {
-    return xs.length > 0 ? Just (xs[xs.length - 1]) : Nothing;
-  }
+
+  function last(foldable) {
+    return Z.reduce (
+      function(acc, a) {
+        return Just(a);
+      },
+      Nothing,
+      foldable
+     );
+    }
+
   _.last = {
-    consts: {},
-    types: [$.Array (a), $Maybe (a)],
+    consts: {f: [Z.Foldable]},
+    types: [f (a), $Maybe (a)],
     impl: last
   };
 
-  //# tail :: Array a -> Maybe (Array a)
+  //# tail :: (Foldable f, Applicative f) => f a -> Maybe (f a)
   //.
-  //. Returns Just all but the first of the given array's elements if the
-  //. array contains at least one element; Nothing otherwise.
+  //. Returns a Just of all but the first of the given Foldable if the
+  //. Foldable contains at least one element; Nothing otherwise.
   //.
   //. ```javascript
   //. > S.tail ([1, 2, 3])
   //. Just ([2, 3])
   //.
   //. > S.tail ([])
+  //.
+  //. > S.tail (Cons ('baz') (Cons ('foo') (Nil)))
+  //. Just (Cons ('foo') (Nil))
```

Comment:
Let's stick with _foo_, _bar_, _baz_ order here as well.

Reply:
🆗

Reply:
This output looks incorrect.

Reply:
🆗